### PR TITLE
NAS-127702 / 24.10 / Recursively mount root dataset during boot if it is not encrypted

### PR DESCRIPTION
**Pull request by usaleem-ix**
Import the pools during boot without mounting any of the datasets. Check on 'feature@encryption' for imported pool. If feature is not active, recursively mount the root datasets, which effectively mounts all datasets present on the pool.

If the encryption feature is active, continue and unlock the datasets.

### CODE REVIEW

**Comment by yocalebo** [comment]
@sonicaj upstream merged changes where we can do `zfs mount -R <root-dataset>` to recursively mount from the parent dataset (including zpool). I've asked Umer to make the changes here at boot. Can you review for me please? Also, I do believe we have some quite complex handling in this area of our code related to encrypted zpools. We can remove that with these changes but that should probably be done in a separate PR.

File: src/middlewared/middlewared/plugins/pool_/import_pool.py [hunk 1]
```diff
@@ -212,13 +212,67 @@ async def import_pool(self, job, data):
 
         return True
 
+    @private
+    def recursive_mount(self, name):
+        cmd = [
+            'zfs', 'mount',
+            '-R',  # recusrive flag
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/pool_/import_pool.py:219]: Comment typo

**Review by sonicaj — APPROVED** [review]
LGTM apart from the typo!

File: src/middlewared/middlewared/plugins/pool_/import_pool.py (hunk 1 above)
Comment (usaleem-ix) [reply src/middlewared/middlewared/plugins/pool_/import_pool.py]: Thanks, typo is fixed.

**Review by sonicaj — APPROVED** [review]
Thanks!

**Pull request merged**